corehq/change_ownership.py

```diff
@@ -0,0 +1,202 @@
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.models import CommCareCase
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.es import CaseSearchES
+from corehq.apps.es.cases import case_type
+import os
+import time
+import math
+
+from casexml.apps.case.mock import CaseBlock
+from corehq.apps.hqcase.utils import submit_case_blocks
+
+# DOMAIN = 'alafiacomm'
+DOMAIN = 'alafiacomm-prod'
+
+success_case_log_file_path = os.path.expanduser('~/script_success.log')
+error_case_log_file_path = os.path.expanduser('~/script_error.log')
+
+BATCH_SIZE = 100
+SKIP_COUNT = 0
+
+def submit_cases(case_blocks):
+    submit_case_blocks(
+        [cb.as_text() for cb in case_blocks],
+        domain=DOMAIN,
+        device_id='system',
+    )
+
+def write_to_log(ids, is_success=True, message=None):
+    if is_success:
+        file_path = success_case_log_file_path
+    else:
+        file_path = error_case_log_file_path
+    with open(file_path, 'a') as log:
+        for id in ids:
+            log.write(
+                'Skipped ' if not is_success else '' \
+                f'{id} - ' \
+                'Reason:' + message if message else ''
+            )
+        log.close()
+
+def process_batch(case_blocks, current_chunk, total_count):
+    batch_start = current_chunk * BATCH_SIZE + 1
+    batch_end = batch_start + len(case_blocks)
+    percentage_done = round((batch_start / total_count) * 100, 2)
+    print(f'Submitting cases {batch_start}-{batch_end}/{total_count} ({percentage_done}%)')
+    try:
+        submit_cases(case_blocks)
+        write_to_log([cb.case_id for cb in case_blocks])
+        return True
+    except Exception as e:
+        write_to_log(
+            [cb.case_id for cb in case_blocks],
+            is_success=False,
+            message=str(e)
+        )
+        return False
+
+### Task 1
+def move_mobile_workers():
+    print("---MOVING MOBILE WORKER LOCATIONS---")
+    RC_NUM_PROP_NAME = 'rc_number'
+    USER_TYPE_PROP_NAME = 'usertype'
+
+    users = CommCareUser.by_domain(DOMAIN)
+    user_count = len(users)
+    success_count = fail_count = skip_count = 0
+    total_time = 0
+    print(f"Total Users to Process: {user_count}")
+    user_to_save = []
+    for idx, user in enumerate(users):
+        start_time = time.time()
+        percentage_done = round((idx / user_count) * 100, 2)
+        print(f'Processing User {idx}/{user_count} ({percentage_done}%)')
+
+        # First make sure that the user type is rc
+        user_data = user.get_user_data(DOMAIN)
+        if user_data[USER_TYPE_PROP_NAME] != 'rc':
+            write_to_log([user.user_id], is_success=False, message="User Type not RC")
+            skip_count += 1
+            continue
+
+        try:
+            # Get a descendant of user location which has the same rc number
```

Comment:
Do we need to look at just the immediate descendants?

Reply:
I can confirm that all the mobile workers need to be moved one level down, so we only need to consider the immediate descendants.